### DISCUSSION TITLE
Github App authentication error

### DISCUSSION THREAD

**Discussion by cizara** [integrations/terraform-provider-github#1797] [Q&A]
Hi, I successfully used this provider to import all our Github organization with a personal token, but now we need to move it to a Github App so we can automate the process. So far, I created a new app with the following permissions:

- Repository permissions - Administration R/W
- Organization permissions - Administration R/W
- Organization permissions - Members R/W

Then installed the app to the Organization, grant access to all repositories, generated a private key and set the empty `app_auth {}` block in the provider.

To test it, I've exported the `GITHUB_APP_*` environment variables as described [here](https://github.com/integrations/terraform-provider-github/tree/main/examples/app_authentication) and running it locally, but I'm facing the following error:
```bash
$ terraform plan

Planning failed. Terraform encountered an error while generating this plan.

╷
│ Error: no decodeable PEM data found
│ 
│   with provider["registry.terraform.io/integrations/github"],
│   on providers.tf line 15, in provider "github":
│   15: provider "github" {
│ 
╵
```
The `GITHUB_APP_PEM_FILE` environment  variable points to the recently downloaded private key.  Did I miss something here?

**Comment by cizara** [comment]
Ok, my bad. So I have to pass the content of the file instead of the path to the file. Using terraform file function solves it for me `pem_file = file(var.pem_file_path)`

**Comment by andrew-shevchyk** [comment]
.. or if you prefer passing credentials in environment variables:
```
export GITHUB_APP_PEM_FILE=$(cat /path/to/file)
```
